### ISSUE TITLE
1.6.0 names golden files differently

### ISSUE THREAD

**Issue by javydreamercsw** [retest/recheck-web#418]
## Describe the Bug

On version 1.6.0 the golden files are named differently. They use a superclass instead of the actual test class rendering old golden files invalid.

### How to Reproduce?

1. Run test with 1.5.1 to create golden files.
2. Repeat same test with 1.6.0.

### Setup

- recheck-web: 1.6.0
- Selenium version: 3.141.59
- Testing framework: TestNG 7.0.0
- OS: [MacOS 14
- Java version: OpenJDK 11

**Comment by javydreamercsw** [comment]
All this was using the AutocheckingRecheckDriver.

**Comment by diba1013** [comment]
This could be related to retest/recheck#444 where we fixed some inner classes not being named properly.

**Comment by diba1013** [comment]
If the automatic system does not work, you could always provide a custom name via
```
RecheckOptions.builder()
    .suiteName( "name" )
    .build()
```

**Comment by javydreamercsw** [comment]
Maybe it's related but now everything uses the base class name instead.

**Comment by diba1013** [comment]
> They use a superclass instead of the actual test class rendering old golden files invalid.

So you mean something like
```
class BaseTest {

    Recheck re = new RecheckImpl();
}

class HomePageTest extends BaseTest {

    @Test
    void doTest() {
        re.check(driver, "foo");
    }
}
```
will create the golden master: `com.example.BaseTest.doTest.foo.recheck` instead of `com.example.HomePageTest.doTest.foo.recheck`.

How do you initialize your `Recheck` instance? Do you provide any custom `NamingStrategy`?

**Comment by javydreamercsw** [comment]
Yes. But I'm using AutocheckingRecheckDriver and it's default naming strategy.

**Comment by diba1013** [comment]
There are two naming strategies:
1. The `NamingStrategy` defines the golden master name, based on the suite, test and check. You can provide custom names for each of them, essentially overwriting the strategy. For more information, please refer to the [docs](https://docs.retest.de/recheck/introduction/usage/).
2. The `AutocheckingCheckNamingStrategy` generates unique names based on the actions (i.e. check names) performed through the `AutocheckingDriver` e.g. "Click on button login".

Note that the `FileNamerStrategy` has been deprecated and replaced by above mentioned `NamingStrategy` and `ProjectLayout`.

With this in mind, and referring to your comment on the default naming strategy, your test should be identified automatically.

**Comment by javydreamercsw** [comment]
That's what I expected but it's not the case.

**Comment by diba1013** [comment]
I just threw together a quick test and can confirm this bug. I will do a more thorough check and bug fix on monday. 

In the meantime, please specify a custom naming strategy.

**Comment by diba1013** [comment]
I did some further investigations...

### Setup

1. Windows 10
2. JUnit 5
3. recheck(-web)-1.6.0 through recheck(-web)-1.5.0
4. Selenium 3.141.59

#### recheck

```
class BaseTest {

	protected Recheck re;

	@BeforeEach
	void setUp() {
		re = new RecheckImpl();
	}

	@AfterEach
	void tearDown() {
		re.cap();
	}
}
```

```
class HomeTest extends BaseTest {

	@Test
	void name() throws Exception {
		re.startTest();
		re.check( "foo", new StringRecheckAdapter(), "bar" );
		re.capTest();
	}
}
```

#### recheck-web

```
class BaseTest {

	protected AutocheckingRecheckDriver driver;

	@BeforeEach
	void setUp() {
		final ChromeDriver driver = new ChromeDriver();
		this.driver = new AutocheckingRecheckDriver( driver );
	}

	@AfterEach
	void tearDown() {
		driver.quit();
	}
}
```

```
class HomeTest extends BaseTest {

	@Test
	void name() throws Exception {
		driver.startTest();

		driver.get( "https://retest.de" );

		final By getStartedButton = By.xpath( "/html/body/main/section[1]/div/div[2]/a" );
		driver.findElement( getStartedButton ).click();

		driver.capTest();
	}
}
```

### Findings

#### recheck

1. 1.6.0: target/test-classes/retest/recheck/de.retest.recheck.**BaseTest**.report
2. 1.5.0: target/test-classes/retest/recheck/de.retest.recheck.**BaseTest**.report

In plain recheck, this has never worked on its own. The fault is not recheck per se.

#### recheck-web

1. 1.6.0: target/test-classes/retest/recheck/de.retest.web.**BaseTest**.report
2. 1.5.1: target/test-classes/retest/recheck/de.retest.web.**HomeTest**.report
3. 1.5.0: target/test-classes/retest/recheck/de.retest.web.**HomeTest**.report

Something changed from 1.5.1 to 1.6.0. I will need to investigate further.

### Conclusion

Put the other way round: A `Recheck` instance always expects to be created within the test class for the automatic system to work. Otherwise a custom suite name (as suggested above) should be provided.

However, the `AutocheckingRecheckDriver` does something special to ensure the actual test class being used. This somehow broke with 1.6.0 and I need to do further investigation on why that happened.

**Comment by javydreamercsw** [comment]
That said is there an equivalent temporary solution for `AutocheckingRecheckDriver`?

**Comment by diba1013** [comment]
I did some further analysis and came to the conclusion that this was caused by retest/recheck#450 where we removed some lazy evaluation of the suite name. This was a fix for our JUnit and TestNG extensions where this caused some problems.

The suite name (a.k.a the Golden Master name) is now evaluated directly when instantiating the options (e.g. through `AutocheckingRecheckDriver#new`), thus getting the `BaseTest` and not the actual `HomeTest`. This is not possible with the stack trace.

We decided, not to reverse this change&mdash;although this unfortunately caused a breaking change for you&mdash;to keep the behavior between recheck and recheck-web the same (i.e. do **not** identify inherit classes). This means, that you must specify the suite name manually. A fix is described in my [comment]( https://github.com/retest/recheck-web/issues/418#issuecomment-552017828) above.
```
RecheckOptions.builder() //
	.suiteName( getClass().getName() ) // This line is now required when using inheritance.
	.build()
```

I will properly document this in our docs.
